### ISSUE TITLE
Error building for .net sample

### ISSUE THREAD

**Issue by alexiordan** [Azure/azure-iot-gateway-sdk#44]
After build.cmd --enable-dotnet-binding
I receive for a number of projects the following error (of course, projects and paths are different) :
```
   633>LINK : fatal error LNK1104: cannot open file 'Files.obj' [c:\gitrepos\azure-iot-gateway-sdk\build\deps\azure-c-shared-utility\testtools\umock_c\tests\umocktypename_ut\umocktypename_ut_dll.vcxproj]
```

**Comment by aribeironovaes** [comment]
Hi @alexiordan, 

Couple of things:
1-What is the version you are trying to build? Is it latest from Develop or latest from Master? (Is it building with out enabling dotnet binding?)
2-If it's latest from develop try to update your submodules. We have changed the submodules name (so it gets shorter), so you may have problems with that). 
3-Have you run build_dotnet.cmd before trying to run build.cmd with dotnet enabled?

Let us know, 

Angelo Ribeiro.

**Comment by alexiordan** [comment]
1. master. Latest from master. Yes, it builds without donet binding. Until it worked, however, I ran into some issues I listed here: [Error following the getting started tutorial ](https://github.com/Azure/azure-iot-gateway-sdk/issues/42)
2. It's latest from master
3. Yes, I have run it. No error given, just a message afterwards: 
`Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.56
'vstest.console' is not recognized as an internal or external command,
operable program or batch file.`

**Comment by aribeironovaes** [comment]
Thanks for confirming @alexiordan, 

I just tested with latest from master and I was able to build, so it must be something different on our environments... Lets try to figure out what it is. 

From your last message I was that it didn't recognize 'vstest.console'as an internal command... Which makes me think you probably are not running a Developer Command Prompt? 

Can you confirm that you are running these scripts from a Developer Command Prompt for VS2015? 

Let us know, 

Angelo Ribeiro

**Comment by aribeironovaes** [comment]
Another thing I would try @alexiordan  is to make sure you have a clean environment. 
Running:
git clean -xdf (make sure you don't have any of your uncommitted files on your enlistment, otherwise you will lose it.)
git submodule update --init --recursive (just to make sure you have the right version from the dependencies, since the error you are getting seems to be in a dependency). 

I will let you know if I can think about something else...

**Comment by alexiordan** [comment]
I was using the Visual Studio 2012 ARM Phone Tools Command Prompt. Thought that they all do the same thing. Now that I think more about it, maybe this was also the root of all evil with my problems encountered in the other issue I mentioned you about.

Now I started the Developer Command Prompt for VS2015 and build_dotnet.cmd run smoothly without that message in the end.

build.cmd --enable-dotnet-binding also worked like a charm (there were some yellow warnings, but overall no errors). Thank you very much for your help. I'm sorry I bothered you with something so stupid.
